configure.ac

```diff
@@ -3246,139 +3246,96 @@ AC_ARG_ENABLE(llvm,[  --enable-llvm	Enable the LLVM back-end], enable_llvm=$enab
 AC_ARG_ENABLE(loadedllvm,[  --enable-loadedllvm	Load the LLVM back-end dynamically], enable_llvm=$enableval && enable_loadedllvm=$enableval, enable_loadedllvm=no)
 AC_ARG_ENABLE(llvm-version-check,[  --enable-llvm-version-check Check that the LLVM matches the version expected by mono], enable_llvm_version_check=$enableval, enable_llvm_version_check=no)
 AC_ARG_ENABLE(llvm-runtime,[  --enable-llvm-runtime	Enable runtime support for llvmonly code], enable_llvm_runtime=$enableval, enable_llvm_runtime=no)
+AC_ARG_ENABLE(llvm-asserts,[  --enable-llvm-asserts Enable llvm asserts (option to LLVM in CMake)], enable_llvm_asserts=$enableval, enable_llvm_asserts=no)
 
 AC_ARG_WITH(llvm, [  --with-llvm=<llvm prefix>    Enable the LLVM back-end], enable_llvm=yes,)
 
 if test "x$enable_llvm" = "xdefault"; then
    enable_llvm=$enable_llvm_default
 fi
 
+MONO_INTERNAL_LLVM=0
```

Comment:
Local variables are lower-cased and do not start by `mono_`. Please rename `MONO_INTERNAL_LLVM` to `internal_llvm`.

Reply:
This isn't used as a local variable, we use it throughout all of the preprocessor/build namespaces. 
```
configure.ac:MONO_INTERNAL_LLVM=0
configure.ac:MONO_INTERNAL_LLVM_ASSERTS=0
configure.ac:         MONO_INTERNAL_LLVM=1
configure.ac:         MONO_INTERNAL_LLVM_ASSERTS=1
configure.ac:AM_CONDITIONAL(MONO_INTERNAL_LLVM, [test "x$MONO_INTERNAL_LLVM" != "x0"])
configure.ac:AC_SUBST(MONO_INTERNAL_LLVM)
configure.ac:AM_CONDITIONAL(MONO_INTERNAL_LLVM_ASSERTS, [test "x$MONO_INTERNAL_LLVM_ASSERTS" != "x0"])
configure.ac:AC_SUBST(MONO_INTERNAL_LLVM_ASSERTS)
configure.ac:	LLVM Back End: $enable_llvm (dynamically loaded: $enable_loadedllvm, built in-tree: $MONO_INTERNAL_LLVM, assertions: $MONO_INTERNAL_LLVM_ASSERTS)
llvm/Makefile.am:if MONO_INTERNAL_LLVM
llvm/Makefile.am:if MONO_INTERNAL_LLVM_ASSERTS
llvm/Makefile.am:endif # MONO_INTERNAL_LLVM_ASSERTS
llvm/Makefile.am:endif # MONO_INTERNAL_LLVM
mono/mini/Makefile.am.in:if MONO_INTERNAL_LLVM 
mono/utils/Makefile.am:if MONO_INTERNAL_LLVM 
```

Reply:
I can make another local variable and then define this one at the end, it seemed easier to search for to define the default value where I compute the non-default value.

Reply:
You are confused between the variables in `configure.ac` and the one in the various `Makefile.am`. It's not because you define a variable in the former that it's going to automatically appear in the latter. Notice the call to `AM_CONDITIONAL` at https://github.com/mono/mono/pull/8792/files/10e86949e0df9db7c2a3a8d360fadfc28d8916b4#diff-67e997bcfdac55191033d57a16d1408aR3331 which is what's going to "declare" the variable in the `Makefile.am`. So as you can also observe, you use the `configure.ac` variable to define the `Makefile.am` variable, and you can use `AM_CONDITIONAL(INTERNAL_LLVM, [test "x$internal_llvm" != "x0"])` to have what you expect.